### ISSUE TITLE
convergence issue

### ISSUE THREAD

**Issue by meysamgolm1360** [xuxingya/tf2crf#25]
Hi,

I created a CNN/LSTM/CRF model for spatiotemporal modeling. But unfortunately it seems that I have convergence problem. The algorithm started with crf_loss: 20.5275 - accuracy: 0.5164 and after 8 epoch ended up with crf_loss: 1.5074 - accuracy: 0.7332. In prediction using the model after 8 epoch, it assigns one class to all inputs. I wonder how I can change the learning rate?

Here is my model just in case:

`    train_data = gen_batch_data(feats_list, label_list, lstm_p)

    # build CNN/LSTM and train it.
    #
    input = Input(shape=(210, 20, 26, 1,))
    
    output = TimeDistributed(Conv2D(16, (3, 3), padding='same'))(input)
    output = Activation('elu')(output)
    output = TimeDistributed(MaxPooling2D(pool_size=(2, 2)))(output)
    output = Dropout(0.2)(output)
    
    output = TimeDistributed(Conv2D(16, (3, 3), padding='same'))(output)
    output = Activation('elu')(output)
    output = TimeDistributed(MaxPooling2D(pool_size=(2, 2)))(output)
    output = Dropout(0.2)(output)
    
    output = TimeDistributed(Conv2D(16, (3, 3), padding='same'))(output)
    output = Activation('elu')(output)
    output = TimeDistributed(MaxPooling2D(pool_size=(2, 2)))(output)
    output = Dropout(0.2)(output)
    
    output =  TimeDistributed(Flatten())(output)
    
    output =  Conv1D(16, 3, padding='same')(output)
    output =  Activation('elu')(output)
    output =  MaxPooling1D(pool_size=10)(output)

    output =  Bidirectional(LSTM(64, return_sequences=True))(output)
    output =  Activation('elu')(output)
    
    output = TimeDistributed(Dense(2, activation='relu'))(output)
    
    crf=CRF(dtype='float32')
    output = crf(output)
    base_model = Model(input, output)
    model = ModelWithCRFLoss(base_model)
    model.compile(optimizer='adam')
    
    # load previous model weights
    #
    if lstm_p['in_model_weights'] != "":
        in_model_weights = nft.get_fullpath(lstm_p['in_model_weights'])
        print (" ... loading previous LSTM model weights from %s ... " \
            % (in_model_weights))
        model.load_weights(in_model_weights)

    # save the model weights after each epoch, if the validation loss decreased
    #
    checkpointer = ModelCheckpoint(filepath = odir + TEMP_WEIGHT_FILE)

    ## Training part..
    #
    model.fit(x=train_data, \
                        epochs=int(lstm_p["epoch_size"]), \
                        verbose=1, \
                        callbacks=[checkpointer], \
                        use_multiprocessing=True, \
                        max_queue_size=10,\
                        workers=12)
    model.summary()`

**Comment by xuxingya** [comment]
`model.compile(tf.keras.optimizers.Adam(learning_rate='5e-5'))`
However, I think ' it assigns one class to all inputs' is abnormal, you
should check your train data first. Is it so imbalanced?

**Comment by meysamgolm1360** [comment]
My hypothesis was right. The issue was related to learning rate. Thank you for your reply.
